policies/documentation-policy.md

```diff
@@ -0,0 +1,122 @@
+
+- descriptions of data structures and their usage
+- provider calls and call backs
+- algorithm descriptions and parameters
+- descriptions of architectural decisions
+
+## <a name="fsgm"></a>Chapter 4: Internal functions, structures, globals and macros
+
+Internal functions and macros are those defined in the `include/crypto`
+and `include/internal` directories.
+
+These should all be documented at the point of implementation.  A comment
+describing their purpose and the input and output arguments and return value
+for functions suffices.
+
+For _trivial_ items, where their operation is obvious from their
+implementation, the documentation requirement is not mandated.  The following
+are generally representative of trivial items, however it is quite
+possible for any of these to be non-trivial in specific instances and
+therefore require documentation:
+
+- OSSL_DISPATCH tables
+- upref functions
+- free functions
+- simple getter/setter functions
+- wrappers for other functions (a function that calls a more recent `_ex`
+  variant or a group of functions that call a common internal routine)
+
+For structures, each of the fields should be commented stating its
+purpose.  Again, a _trivial_ exception applies where the purpose is
+obvious.  Some representative examples:
+
+- `OSSL_LIB_CTX *ctx;` where there is only one library context referenced in
+  the structure.
+- `struct *next;` in a linked list implementation.
+- `CRYPTO_REF_COUNT refcnt;`
+
+## Chapter 5: Static functions and globals and local structures and macros
+
+These are functions, structures, globals and macros local to a specific
+C file or defined for a single directory as part of a local .h file.
+
+These should all be documented at the point of implementation.  Follow the
+rules and exceptions listed [above](#fsgm).  In some cases slightly more
+leniency with respect to _trivial_ can be tolerated.
+
+## Chapter 6: Code comments
+
+Comments in code should explain what the algorithm is doing and why.
+It should not parrot the effect of each statement.  As the complexity of
+the code increases, the size and detail of comments should also increase.
+
+## Chapter 7: Assembly code
+
+Assembly code should include a good description of the algorithm and
+approach being used.  This should be followed by a performance comparison
+and then the assembly code itself.  The assembly code should be well
+commented, but it isn't necessary to comment every line.  A comment
+describing each block of code suffices.
+
+There are no _trivial_ exceptions for assembly code.
+
+## Chapter 8: Configure options
+
+New options added to the configuration scripts must be documented in the
+`INSTALL.md` file.
+
+## Chapter 9: Changes and news
+
+Significant modifications should be documented in the `CHANGES.md` file.
+
+Very significant features and changes should be documented in the `NEWS.md`
+file.
+
+In both cases, the added note should be short and to the point.
```

Comment:
This seems redundant, with `NEWS.md` being an abridged version of `CHANGES.md`.

I would suggest discarding `NEWS.md` and just reference `CHANGES.md`. "Very significant" features in `CHANGES.md` can be highlighted in some manner to identify them as such.

Reply:
I think both files have their use but I do see your point.
Possibly something for future discussion.